src/tools/crawler/rpc.rs

```diff
@@ -34,9 +38,24 @@ pub async fn initialize_rpc_server(rpc_addr: SocketAddr, rpc_context: RpcContext
 fn create_rpc_module(rpc_context: RpcContext) -> RpcModule<RpcContext> {
     let mut module = RpcModule::new(rpc_context);
     module
-        .register_method("getmetrics", |_, rpc_context| {
+        .register_method("getmetrics", |params, rpc_context| {
+            let report_params = params.parse::<ReportParams>()?;
+            if let Some(path) = report_params.file {
+                let content = serde_json::to_string(rpc_context.0.lock().unwrap().deref())?;
+                // TODO: consider some checks against directory traversal
```

Comment:
To check if someone is not trying to put parent directory somewhere we can just use:
```
if pathBuf.components().into_iter().any(|p| p == Component::ParentDir) {
   // directory traversal here...
}
```
Or check if there are any other '/' symbols (especially on the beginning).

Reply:
Yes, disallowing any relative path is one way. Also, we could do `canonicalize` and check if the final path is somewhere we allow to be. Or, take only filename part and dump the file either in /tmp or current dir.
Let's wait and see how this file is going to be read, then we'll figure out how to do this safely.